rl/function_approx.py

```diff
@@ -36,13 +47,22 @@ def within(self, other: FunctionApprox[X], tolerance: float) -> bool:
         '''
         pass
 
-    @staticmethod
-    def converged(iterator: Iterator[FunctionApprox[X]],
-                  tolerance: float = 0.0001) -> FunctionApprox[X]:
-        def done(a, b):
-            return a.within(b, tolerance)
+    def rmse(
+        self,
+        xy_seq: Sequence[Tuple[X, float]]
+    ) -> float:
+        x_seq, y_seq = zip(*xy_seq)
+        errors: np.ndarray = self.evaluate(x_seq) - np.array(y_seq)
+        return np.sqrt(np.mean(errors * errors))
 
-        return iterate.converged(iterator, done=done)
+    def iterate_updates(
+        self,
+        xy_seq_stream: Iterator[Sequence[Tuple[X, float]]]
+    ) -> Iterator[FunctionApprox[X]]:
+        func_approx: FunctionApprox[X] = self
+        for xy_seq in xy_seq_stream:
+            yield func_approx
+            func_approx = func_approx.update(xy_seq)
```

Comment:
The for loop design is totally reasonable, but you could also write this pretty elegantly with the `accumulate` function. Something like:

```
return itertools.accumulate(xy_seq_stream, lambda fa, xy: fa.update(xy), initial=self)
```

Reply:
Much nicer! Made the change